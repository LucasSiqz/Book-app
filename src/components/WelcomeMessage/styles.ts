import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;

    h2 {
      font-size: ${theme.font.sizes.xxlarge};
    }

    h2:first-child {
      color: ${theme.colors.darkGray};
    }
  `}
`
export const Name = styled.h2`
  ${({ theme }) => css`
    margin: 0 1rem;
    color: ${theme.colors.pink};
  `}
`
