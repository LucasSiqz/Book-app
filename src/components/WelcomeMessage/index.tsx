import * as S from './styles'

export type WelcomeMessageProps = {
  name: string
}

const WelcomeMessage = ({ name = 'Mehmed Al Fatih' }: WelcomeMessageProps) => (
  <S.Wrapper>
    <h2>Hi,</h2>
    <S.Name>{name}</S.Name>
    <h2>ğŸ‘‹</h2>
  </S.Wrapper>
)

export default WelcomeMessage
