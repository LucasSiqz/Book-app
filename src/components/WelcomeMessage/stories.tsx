import { Story, Meta } from '@storybook/react/types-6-0'
import WelcomeMessage, { WelcomeMessageProps } from '.'

export default {
  title: 'WelcomeMessage',
  component: WelcomeMessage,
  args: {
    name: 'Mehmed Al Fatih'
  }
} as Meta

export const Default: Story<WelcomeMessageProps> = (args) => (
  <WelcomeMessage {...args} />
)
