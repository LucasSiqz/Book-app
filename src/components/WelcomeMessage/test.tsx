import { render, screen } from 'utils/test-utils'

import WelcomeMessage from '.'

const props = {
  name: 'Lucas'
}

describe('<WelcomeMessage />', () => {
  it('should render correctly', () => {
    const { container } = render(<WelcomeMessage {...props} />)

    expect(screen.getByRole('heading', { name: /Lucas/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
