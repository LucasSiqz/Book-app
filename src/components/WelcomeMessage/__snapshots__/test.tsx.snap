// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeMessage /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 h2 {
  font-size: 2.4rem;
}

.c0 h2:first-child {
  color: #3F4043;
}

.c1 {
  margin: 0 1rem;
  color: #FF6978;
}

<div
  class="c0"
>
  <h2>
    Hi,
  </h2>
  <h2
    class="c1"
  >
    Lucas
  </h2>
  <h2>
    ðŸ‘‹
  </h2>
</div>
`
